### ISSUE TITLE
Middlewares are applied in last-in-first-out order

### ISSUE THREAD

**Issue by AmaranthineCodices** [Roblox/rodux#28]
This is a tripup in how middlewares are applied. This ordering of middlewares:

```
{ middlewareA, middlewareB, middlewareC }
```

is invoked like this:

```
dispatch(middlewareA(middlewareB(middlewareC(action))))
```

in short, they're being applied in reverse order. This is not intuitive behavior; middlewares should be applied in first-in-first-out order, so the middlewares above would be invoked like:

```
dispatch(middlewareC(middlewareB(middlewareA(action))))
```

**Comment by LPGhatguy** [comment]
I feel like both orders are still pretty confusing. Someone is always going to feel that they're backwards!

The current order lines up with functional composition syntax in mathematics, like `f ∘ g` being `f(g(x))`.

**Comment by LPGhatguy** [comment]
We fixed this in #29!

Everyone was very confused with regards to this, but you were totally right.
